### PR TITLE
Publish 2024-04-01 version of Text-to-Speech Batch Synthesis API spec

### PR DESCRIPTION
**Pull request by Ling-Cao**
# Data Plane API - Pull Request

The PR is created based on the updates in the private branch with https://github.com/Azure/azure-rest-api-specs-pr/issues/14489 which has already been reviewed and approved. 

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue: https://github.com/Azure/azure-rest-api-specs-pr/issues/14999

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [x] GA release

### Change Scope

Check [this comparation link](https://github.com/Ling-Cao/azure-rest-api-specs/compare/caoling/batchsynthesis-v2024-04-01...Ling-Cao:azure-rest-api-specs:caoling/batchsynthesis-v2024-04-01-latest#diff-a5e1065bca9807191d811ed1c6eead81500f0c18ca4428b0250d8fdec4234880) to see the difference of this PR and last approved changes.

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

**Comment by mikekistler** [comment]
@heaths @weidongxu-microsoft Please take a quick look at this PR. We held a API review for this back in Oct of last year and both of you were in that meeting. Here's a link to the API review issue:

https://github.com/Azure/azure-rest-api-specs-pr/issues/14999

You can see there that in that review we decided:
> This looks good, just a few issues in the above comments and the PR.

I will review and would like at least one more review before I add the signed off label.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+          "description": "The number of billable characters for AOAI HD voices in the input file.",
+          "type": "integer"
+        },
+        "personalVoiceCharacters": {
+          "format": "int64",
+          "description": "The number of billable characters for personal voices in the input file.",
+          "type": "integer"
+        }
+      },
+      "readOnly": true
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json:376]: The `readOnly` attribute should only be specified in schema property definitions.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
+        "code": {
+          "description": "Top level error code",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message",
+          "type": "string"
+        }
+      },
+      "readOnly": true
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json:424]: The `readOnly` attribute should only be specified in schema property definitions.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json [hunk 3]
```diff
@@ -0,0 +1,779 @@
...
+        "result": {
+          "description": "sas url of the result file. It would be relative path when \"destinationContainerUrl\" is specified.",
+          "type": "string"
+        },
+        "summary": {
+          "description": "When BatchSynthesisProperties.decompressOutputFiles is true, job summary file path will be provided here.",
+          "type": "string"
+        }
+      },
+      "readOnly": true
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json:453]: The `readOnly` attribute should only be specified in schema property definitions.

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

I left a few very minor comments -- not necessary to fix but would fix a discrepancy with the OpenAPI specification.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json (hunk 1 above)
Comment (weidongxu-microsoft) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json:376]: Agree. I think the `readOnly` should be on parent schema `billingDetails` property?

Similar for the `BatchSynthesisError`.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json [hunk 4]
```diff
@@ -492,7 +491,7 @@
           "format": "int32"
         },
         "outputFormat": {
-          "description": "The requested audio output format.",
+          "description": "The requested audio output format. Reference https://learn.microsoft.com/en-us/azure/ai-services/speech-service/batch-synthesis-properties#batch-synthesis-properties.",
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json:494]: same here, no "en-us"

Comment (Ling-Cao) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json]: Thank you and fixed.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json [hunk 5]
```diff
@@ -0,0 +1,779 @@
...
+            "schema": {
+              "$ref": "#/definitions/PaginatedBatchSynthesis"
+            }
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
```
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json]: Make sure you have the header in backend.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json [hunk 6]
```diff
@@ -0,0 +1,779 @@
...
+      "properties": {
+        "timeToLiveInHours": {
+          "description": "How long the batch task will be kept in the system after it has completed. Once the batch task reaches the time to live after completion (succeeded or failed) it will be automatically deleted. The longest supported duration is 744 hours (31 days). The default value is 744 hours. ",
+          "type": "integer",
+          "format": "int32"
+        },
+        "outputFormat": {
+          "description": "The requested audio output format.",
+          "type": "string"
+        },
```
[the remark this answers is not in the document]
Comment (Ling-Cao) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json]: Yes, user can set one, I added the reference document link here.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json [hunk 7]
```diff
@@ -0,0 +1,779 @@
...
+    },
+    "BatchSynthesisConfig": {
+      "description": "Text-to-speech configuration for plain text input.",
+      "required": [
+        "voice"
+      ],
+      "type": "object",
+      "properties": {
+        "voice": {
+          "description": "Voice name for PlainText input. You can find platform voice list here https://learn.microsoft.com/en-us/azure/ai-services/speech-service/language-support?tabs=tts.",
```
[the remark this answers is not in the document]
Comment (Ling-Cao) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json]: Thank you and fixed.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json (hunk 3 above)
Comment (Ling-Cao) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json]: Thank you and fixed.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json (hunk 2 above)
Comment (Ling-Cao) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json]: Thank you and fixed.

**Review by weidongxu-microsoft — APPROVED** [review]
I see author fixed most of my comments.

There is a naming of SSML enum value (the change of which would have impact on backend), but I assume that won't be a blocker.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json (hunk 1 above)
Comment (Ling-Cao) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json]: Thank you and fixed.

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json [hunk 8]
```diff
@@ -0,0 +1,779 @@
...
+        "name": "Status",
+        "modelAsString": true
+      },
+      "readOnly": true
+    },
+    "InputKind": {
+      "description": "Input text kind.",
+      "enum": [
+        "PlainText",
+        "SSML"
```
[the remark this answers is not in the document]
Comment (Ling-Cao) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json:719]: The full wording might be too long as this property is required in `payload.InputKind` to create a job.

**Review by ms-henglu — APPROVED** [review]

File: specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (Ling-Cao) [reply specification/cognitiveservices/data-plane/Speech/BatchTextToSpeech/stable/2024-04-01/batchsynthesis.json:719]: Thank you, we accepted PascalCase `Ssml` and updated.

**Pull request merged**